Composer/packages/tools/language-servers/language-generation/src/LGServer.ts

```
@@ -0,0 +1,355 @@
+          codeActionProvider: false,
+          completionProvider: {
+            resolveProvider: true,
+          },
+          hoverProvider: true,
+          foldingRangeProvider: false,
+        },
+      };
+    });
+    this.connection.onCompletion(params => this.completion(params));
+    this.connection.onHover(params => this.hover(params));
+
+    this.connection.onRequest((method, params) => {
+      if (InitializeDocumentsMethodName === method) {
+        const { uri, content, template } = params;
+        this.LGDocuments.push({ uri, content, template });
+        // run diagnostic
+        const textDocument = this.documents.get(uri);
+        if (textDocument) {
+          setTimeout(() => {
+            this.validate(textDocument);
+          }, 500);
+        }
+      }
+    });
+  }
+
+  start() {
+    this.connection.listen();
+  }
+
+  protected getLGDocument(document: TextDocument): LGDocument | undefined {
+    const LGDocument = this.LGDocuments.find(item => item.uri === document.uri);
+    return LGDocument;
+  }
+  protected getLGDocumentContent(document: TextDocument): string {
+    const LGDocument = this.LGDocuments.find(item => item.uri === document.uri);
+    let text = document.getText();
+    if (LGDocument) {
+      // concat new content
+      const { content, template } = LGDocument;
+      try {
+        text = updateTemplateInContent(content, template);
+        // eslint-disable-next-line no-empty
+      } catch (error) {}
+    }
+    return text;
+  }
+
+  protected hover(params: TextDocumentPositionParams): Thenable<Hover | null> {
+    const document = this.documents.get(params.textDocument.uri);
+    if (!document) {
+      return Promise.resolve(null);
+    }
+    const text = this.getLGDocumentContent(document);
+    let templates: LGTemplate[] = [];
+    try {
+      const lgResources = getLGResources(text);
+      templates = lgResources.Templates;
+      // eslint-disable-next-line no-empty
+    } catch (error) {}
```

Try not to silence errors.

In this case, if LG document somehow corrupted or the parser has coding errors. `templates` will be empty. But in fact there should be content inside.

That means, this could accidentally destroy user data.

---

refactor it, found trycatch is not necessary